ria_remote/remote.py

```diff
@@ -224,32 +230,27 @@ def __init__(self, host):
           on.
         """
 
-        try:
-
-            from datalad.support.sshconnector import SSHManager
-            # connection manager -- we don't have to keep it around, I think
-            self.sshmanager = SSHManager()
-            # the connection to the remote
-            # we don't open it yet, not yet clear if needed
-            self.ssh = self.sshmanager.get_connection(
-                host,
-                use_remote_annex_bundle=False,
-            )
-            self.ssh.open()
-            # open a remote shell
-            cmd = ['ssh'] + self.ssh._ssh_args + [self.ssh.sshri.as_str()]
-            self.shell = subprocess.Popen(cmd, stderr=subprocess.DEVNULL, stdout=subprocess.PIPE, stdin=subprocess.PIPE)
-            # swallow login message(s):
-            self.shell.stdin.write(b"echo RIA-REMOTE-LOGIN-END\n")
-            self.shell.stdin.flush()
-            while True:
-                line = self.shell.stdout.readline()
-                if line == b"RIA-REMOTE-LOGIN-END\n":
-                    break
-            # TODO: Same for stderr?
-
-        except Exception as e:
-            raise RemoteError(str(e))
+        from datalad.support.sshconnector import SSHManager
+        # connection manager -- we don't have to keep it around, I think
+        self.sshmanager = SSHManager()
+        # the connection to the remote
+        # we don't open it yet, not yet clear if needed
+        self.ssh = self.sshmanager.get_connection(
+            host,
+            use_remote_annex_bundle=False,
+        )
+        self.ssh.open()
+        # open a remote shell
+        cmd = ['ssh'] + self.ssh._ssh_args + [self.ssh.sshri.as_str()]
+        self.shell = subprocess.Popen(cmd, stderr=subprocess.DEVNULL, stdout=subprocess.PIPE, stdin=subprocess.PIPE)
+        # swallow login message(s):
+        self.shell.stdin.write(b"echo RIA-REMOTE-LOGIN-END\n")
+        self.shell.stdin.flush()
+        while True:
+            line = self.shell.stdout.readline()
+            if line == b"RIA-REMOTE-LOGIN-END\n":
+                break
+        # TODO: Same for stderr?
```

Comment:
We don't need a RemoteError?

Reply:
Nope. That's the point of the decorator. We can let exceptions through from within the *IO classes. Will be catched by the decorated `RIARemote` methods, that actually face the outside world via special remote protocol. In case of this `__init__` it's `RIARemote.prepare()`.

Reply:
Cool!